README.md

```diff
@@ -5,4 +5,4 @@
 [![](https://img.shields.io/badge/docs-stable-blue.svg)](https://beacon-biosignals.github.io/TimeSpans.jl/stable)
 [![](https://img.shields.io/badge/docs-dev-blue.svg)](https://beacon-biosignals.github.io/TimeSpans.jl/dev)
 
-TimeSpans.jl provides a simple `TimeSpan` type for representing a continuous span between two points in time, along with generic utility functions for common operations on `TimeSpan`-like types. Importantly, the package exposes a minimal interface (`TimeSpans.start` and `TimeSpans.stop`) that any type can implement to enable support for the TimeSpans API.
+TimeSpans.jl provides a simple `TimeSpan` type for representing a continuous span between two points in time, along with generic utility functions for common operations on `TimeSpan`-like types. Importantly, the package exposes a minimal interface (`TimeSpans.start` and `TimeSpans.stop`) that any type can implement to enable support for the TimeSpans API. NamedTuples of the form `(;start=Nanosecond(1), stop=Nanosecond(2))` are supported in many cases (all those where the function belongs to `TimeSpans`).
```

Comment:
```suggestion
TimeSpans.jl provides a simple `TimeSpan` type for representing a continuous span between two points in time, along with generic utility functions for common operations on `TimeSpan`-like types. Importantly, the package exposes a minimal interface (`TimeSpans.start` and `TimeSpans.stop`) that any type can implement to enable support for the TimeSpans API. In addition to `TimeSpan`, this package implements the TimeSpans API for `Period`s and `NamedTuple`s that have `:start` and `:stop` keys.
```

Some notes that apply to the PR as a whole, motivated by the suggested change above:
- It should be possible to support arbitrary `NamedTuple`s that have `:start` and `:stop` at minimum, possibly as part of a larger set of keys, e.g. `(; start=Second(1), stop=Second(5), event="explosion")`. AFAICT the only real tradeoff is that you get an `ErrorException` rather than a `MethodError` if you're missing `:start` or `:stop`.
- IMO, the `:start` and `:stop` keys needn't be `Nanosecond`-valued or even `Period`-valued; in the same manner that e.g. `TimeSpan(0, 10)` works, we could convert to `Nanosecond` as needed by defining
   ```julia
   start(nt::NamedTuple) = Nanosecond(nt.start)
   stop(nt::NamedTuple) = Nanosecond(nt.stop)
   ```
   which should provide the same set of guarantees.
- ...actually, we could even define an untyped fallback method that does exactly the above, which would cover `NamedTuple`s for free. Hmmm.

Also note that there should be a space after `;` in the named tuple literals. (Won't comment inline for all.)

Reply:
I definitely see the argument for expanding the definition to be more generic. My goal here was mostly to simply copy the functionality as implemented in Onda.jl. (I'll fix the stylistic changes). That seems like a reasonable scope for this PR.